### PR TITLE
feat: implement filtering by status in new ui

### PR DESCRIPTION
**Pull request by shadowusr**
Example report: https://nda.ya.ru/t/VLqVLf-d77fFpu

⚠️ This PR is based on #590, only the last commit needs review (`feat: implement filtering by status in new ui`).

### CODE REVIEW

File: lib/static/new-ui/features/suites/components/TestStatusFilter/index.module.css [hunk 1]
```diff
@@ -0,0 +1,9 @@
+.test-status-filter-option {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+}
+
+    .test-status-filter-option__count {
```
Comment (sipayRT) [inline lib/static/new-ui/features/suites/components/TestStatusFilter/index.module.css:7]: ←

Comment (sipayRT) [reply lib/static/new-ui/features/suites/components/TestStatusFilter/index.module.css:7]: or you can use stylus:

```css
.test-status-filter-option {

    &__count {
    
    }
}
```

File: lib/static/new-ui/features/suites/components/TestStatusFilter/index.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+interface TestStatusFilterProps {
+    statusCounts: StatusCounts;
+    actions: typeof actions;
+    viewMode: ViewMode;
+}
+
+function TestStatusFilterInternal({statusCounts, actions, viewMode}: TestStatusFilterProps): ReactNode {
+    return <RadioButton width={'max'} onChange={(e): void => void actions.changeViewMode(e.target.value)} value={viewMode}>
+        <RadioButton.Option value={ViewMode.ALL} content={<TestStatusFilterOption status={'total'} count={statusCounts.total}/>} />
```
Comment (sipayRT) [inline lib/static/new-ui/features/suites/components/TestStatusFilter/index.tsx:32]: i would like to add a title for each option (but i'm not sure that it's possible for this component)

File: lib/static/new-ui/features/suites/components/TestStatusFilter/selectors.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+export const getStatusCounts = createSelector(
+    [getResults, getAllBrowserIds],
+    (results, browserIds) => {
+        const latestAttempts: Record<string, {attempt: number; status: string, timestamp: number}> = {};
+        const retriedTests = new Set<string>();
+
+        let retries = 0;
+        Object.values(results).forEach(result => {
+            const {parentId: testId, attempt, status, timestamp} = result;
+            if (attempt > 0) {
```
Comment (sipayRT) [inline lib/static/new-ui/features/suites/components/TestStatusFilter/selectors.ts:23]: can we check it by `status === RETRIED`?

File: lib/static/new-ui/features/suites/components/TestStatusFilter/selectors.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            success: 0,
+            fail: 0,
+            skipped: 0,
+            total: browserIds.length,
+            retried: retriedTests.size,
+            retries,
+            idle: 0
+        };
+
+        Object.values(latestAttempts).forEach(({status}) => {
```
Comment (sipayRT) [inline lib/static/new-ui/features/suites/components/TestStatusFilter/selectors.ts:44]: why can't we do this for all results without `retried` filtering?

**Review by sipayRT — APPROVED** [review]

File: lib/static/new-ui/features/suites/components/TestStatusFilter/selectors.ts (hunk 3 above)
Comment (shadowusr) [reply lib/static/new-ui/features/suites/components/TestStatusFilter/selectors.ts:23]: No, status stores an actual test status like fail, error or success. There's no such thing as "retried" status in sqlite db or anywhere else.

File: lib/static/new-ui/features/suites/components/TestStatusFilter/selectors.ts (hunk 4 above)
Comment (shadowusr) [reply lib/static/new-ui/features/suites/components/TestStatusFilter/selectors.ts:44]: We need to compute final test results before counting. Results are not linked between each other and this is the only way to determine their final status.

File: lib/static/new-ui/features/suites/components/TestStatusFilter/index.tsx (hunk 2 above)
Comment (shadowusr) [reply lib/static/new-ui/features/suites/components/TestStatusFilter/index.tsx:32]: Done!
Looks like this now:

**Pull request merged**